src/transformers/models/encodec/feature_extraction_encodec.py

```diff
@@ -0,0 +1,200 @@
+            return_attention_mask (`bool`, *optional*):
+                Whether to return the attention mask. If left to the default, will return the attention mask according
+                to the specific feature_extractor's default.
+
+                [What are attention masks?](../glossary#attention-mask)
+
+            return_tensors (`str` or [`~utils.TensorType`], *optional*):
+                If set, will return tensors instead of list of python integers. Acceptable values are:
+
+                - `'tf'`: Return TensorFlow `tf.constant` objects.
+                - `'pt'`: Return PyTorch `torch.Tensor` objects.
+                - `'np'`: Return Numpy `np.ndarray` objects.
+            sampling_rate (`int`, *optional*):
+                The sampling rate at which the `audio` input was sampled. It is strongly recommended
+                to pass `sampling_rate` at the forward call to prevent silent errors.
+        """
+        if sampling_rate is not None:
+            if sampling_rate != self.sampling_rate:
+                raise ValueError(
+                    f"The model corresponding to this feature extractor: {self} was trained using a sampling rate of"
+                    f" {self.sampling_rate}. Please make sure that the provided audio input was sampled with"
+                    f" {self.sampling_rate} and not {sampling_rate}."
+                )
+        else:
+            logger.warning(
+                "It is strongly recommended to pass the ``sampling_rate`` argument to this function. "
+                "Failing to do so can result in silent errors that might be hard to debug."
+            )
+
+        is_batched = bool(
+            isinstance(audio, (list, tuple))
+            and (isinstance(audio[0], np.ndarray) or isinstance(audio[0], (tuple, list)))
+        )
+
+        if is_batched:
+            audio = [np.asarray(audio, dtype=np.float32).T for audio in audio]
+        elif not is_batched and not isinstance(audio, np.ndarray):
+            audio = np.asarray(audio, dtype=np.float32)
+        elif isinstance(audio, np.ndarray) and audio.dtype is np.dtype(np.float64):
+            audio = audio.astype(np.float32)
+
+        # always return batch
+        if not is_batched:
+            audio = [np.asarray(audio).T]
+
+        # verify inputs are valid
+        for idx, example in enumerate(audio):
+            if example.ndim == 1:
+                example = example[..., None]
+            if example.ndim > 2:
+                raise ValueError(f"Expected input shape (channels, length) but got shape {example.T.shape}")
+            if self.feature_size == 1 and example.shape[-1] != 1:
+                raise ValueError(f"Expected mono audio but example has {example.shape[-1]} channels")
+            if self.feature_size == 2 and example.shape[-1] != 2:
+                raise ValueError(f"Expected stereo audio but example has {example.shape[-1]} channels")
```

Comment:
nice error - wonder if we should though instead throw a warning and just do conversion ourselves:
```py
stereo_audio = np.concatenate([mono_audio, mono_audio], axis=0)
```

Reply:
Both ok to leave as is for now or to do automatic conversion. Maybe to begin with let's just leave it to be faster?

Reply:
Agreed!